src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/resource_providers/batch_provider.py

```diff
@@ -0,0 +1,71 @@
+from azure.mgmt.batch import BatchManagementClient
+from azure.mgmt.batch.models import PrivateLinkServiceConnectionState, PrivateLinkServiceConnectionStatus
+from knack.log import get_logger
+from . import PrivateEndpointClient
+
+logger = get_logger(__name__)
+
+
+class BatchPrivateEndpointClient(PrivateEndpointClient):
+    @staticmethod
+    def _update_private_endpoint_connection_status(client, resource_group_name,
+                                                   account_name,
+                                                   private_endpoint_connection_name,
+                                                   is_approved=True,
+                                                   description=None):
+        new_status = PrivateLinkServiceConnectionState(
+            status=PrivateLinkServiceConnectionStatus.approved,
+            description=description) if is_approved else \
+            PrivateLinkServiceConnectionState(
+                status=PrivateLinkServiceConnectionStatus.rejected,
+                description=description)
+
+        return client.update(
+            resource_group_name=resource_group_name,
+            account_name=account_name,
+            private_endpoint_connection_name=private_endpoint_connection_name,
+            private_link_service_connection_state=new_status)
+
+    def list_private_link_resource(self, cmd, resource_group_name, name):
+        client = get_mgmt_service_client(cmd.cli_ctx, BatchManagementClient).private_link_resource
+        return client.list_by_batch_account(resource_group_name, name)
+
+    def approve_private_endpoint_connection(self, cmd, resource_group_name,
+                                            resource_name, name, approval_description=None):
+        client = get_mgmt_service_client(cmd.cli_ctx, BatchManagementClient).private_endpoint_connection
+        return self._update_private_endpoint_connection_status(
+            client=client,
+            resource_group_name=resource_group_name,
+            account_name=resource_name,
+            private_endpoint_connection_name=name,
+            is_approved=True,
+            description=approval_description)
+
+    def reject_private_endpoint_connection(self, cmd, resource_group_name,
+                                           resource_name, name, rejection_description=None):
+        client = get_mgmt_service_client(cmd.cli_ctx, BatchManagementClient).private_endpoint_connection
+        return self._update_private_endpoint_connection_status(
+            client=client,
+            resource_group_name=resource_group_name,
+            account_name=resource_name,
+            private_endpoint_connection_name=name,
+            is_approved=False,
+            description=rejection_description)
+
+    def remove_private_endpoint_connection(self, cmd, resource_group_name, resource_name, name):
+        logger.error("Microsoft.Batch/batchAccounts does not currently support deleting private endpoint "
```

Comment:
Is it the reason that you don't use generic implementation?

Reply:
Since our service is HOBO(hosted on behalf of), meaning that we host VMSS on behalf of the customer we had to implement our private endpoint API's ourselves, and we do not support this operation. The Batch side API got sign off from networking, but since we are in this special scenario, and since you wanted us to conform to the generic cmdlets, we need the separate provider. 

Like I said this was a JEDI commitment for 5/15 that we allowed to slip due to the regression we found but this needs to go out as part of this release cadence. Batch only supports https://docs.microsoft.com/en-us/rest/api/batchmanagement/privateendpointconnection and this was approved by Networking.

Reply:
Oh sorry read this as `Is there` rather than `is this` (don't know which you meant). That plus some of our models in our API are named different (specifically for update)